code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -27,6 +27,7 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	var/volume = CELL_VOLUME //liters
 	var/last_share = 0
 	var/list/reaction_results
+	var/static/list/nonreactive_gases = list(typecacheof(/datum/gas/oxygen, /datum/gas/nitrogen, /datum/gas/carbon_dioxide)) // These gasses cannot react amongst themselves

```

Comment:
2 things.

1 Move this list down into the react proc
2 It shouldn't be a hard coded list, we have the functionality to determine what gases are required for reactions (/datum/gas_reaction/init_reqs()), use that to dynamically build the typecache list using a helper proc